### ISSUE TITLE
Eslint 9 fails with legacy config .eslintrc

### ISSUE THREAD

**Issue by MikeMcC399** [cypress-io/eslint-plugin-cypress#156]
## Issue

Linting with an [Eslint deprecated Configuration File](https://eslint.org/docs/latest/use/configure/configuration-files-deprecated) fails when using the environment variable setting `ESLINT_USE_FLAT_CONFIG` = `false` and the repository being tested uses the configuration file

`.eslintrc`.

## Versions

`eslint@9.0.0`
`eslint-plugin-cypress@2.15.1`
Ubuntu `22.04.4` LTS
Node.js `20.12.1` LTS

## Steps to reproduce

Ubuntu `22.04.4` LTS, Node.js `20.12.1` LTS

```shell
git clone --branch test/eslint-9-cypress https://github.com/MikeMcC399/cypress-example-kitchensink
cd cypress-example-kitchensink
npm ci
export ESLINT_USE_FLAT_CONFIG=false
npm run lint
```

## Logs

```text
$ npm run lint

> cypress-example-kitchensink@0.0.0-development lint
> eslint --fix cypress app/assets/js/scripts.js


Oops! Something went wrong! :(

ESLint: 9.0.0

TypeError: context.getAncestors is not a function
Occurred while linting /home/mike/github/tmp/cypress-example-kitchensink/cypress/e2e/1-getting-started/todo.cy.js:20
Rule: "cypress/no-async-tests"
    at Identifier (/home/mike/github/tmp/cypress-example-kitchensink/node_modules/eslint-plugin-cypress/lib/rules/no-async-tests.js:32:37)
    at ruleErrorHandler (/home/mike/github/tmp/cypress-example-kitchensink/node_modules/eslint/lib/linter/linter.js:1145:48)
    at /home/mike/github/tmp/cypress-example-kitchensink/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/home/mike/github/tmp/cypress-example-kitchensink/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/home/mike/github/tmp/cypress-example-kitchensink/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (/home/mike/github/tmp/cypress-example-kitchensink/node_modules/eslint/lib/linter/node-event-generator.js:326:22)
    at NodeEventGenerator.enterNode (/home/mike/github/tmp/cypress-example-kitchensink/node_modules/eslint/lib/linter/node-event-generator.js:340:14)
    at runRules (/home/mike/github/tmp/cypress-example-kitchensink/node_modules/eslint/lib/linter/linter.js:1184:40)
    at Linter._verifyWithoutProcessors (/home/mike/github/tmp/cypress-example-kitchensink/node_modules/eslint/lib/linter/linter.js:1453:31)
(node:5779) ESLintRCWarning: You are using an eslintrc configuration file, which is deprecated and support will be removed in v10.0.0. Please migrate to an eslint.config.js file. See https://eslint.org/docs/latest/use/configure/migration-guide for details.
(Use `node --trace-warnings ...` to show where the warning was created)
```

## Related

- https://github.com/cypress-io/eslint-plugin-cypress/issues/153
- https://github.com/cypress-io/eslint-plugin-cypress/issues/146
- https://github.com/cypress-io/eslint-plugin-cypress/issues/154
- https://github.com/cypress-io/eslint-plugin-cypress/issues/155

## References

- [Eslint > Configuration Files (Deprecated)](https://eslint.org/docs/latest/use/configure/configuration-files-deprecated)

**Comment by brettz9** [comment]
Besides the flat files, ESLint 9 also changed some APIs. In this case, as can be seen in the error message, it is that `context.getAncestors` has been dropped. See https://eslint.org/blog/2023/09/preparing-custom-rules-eslint-v9/#context.getancestors()

**Comment by MikeMcC399** [comment]
Thanks @brettz9 !

- I was struggling a little about how to report the different migration issues.  I've noted this also in https://github.com/cypress-io/eslint-plugin-cypress/issues/154.

- There is a similar issue also for the related mocha plugin https://github.com/lo1tuma/eslint-plugin-mocha/issues/352

**Comment by MikeMcC399** [comment]
- There is a great tracking list on https://github.com/eslint/eslint/issues/18093 showing Eslint 9 compatibility progress.

**Comment by MikeMcC399** [comment]
The plugin should prevent installation with ESLint `9.x` until it is compatible with this version. This may need adding an [engines](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#engines) constraint to `package.json`.

**Comment by MikeMcC399** [comment]
- See also related issue #175

**Comment by MikeMcC399** [comment]
- This issue is resolved by PR #176 which restricts the use of the plugin to ESLint `v7` and `v8`

- The enhancement request to support ESLint `v9` is logged under issue #155

- The underlying API deprecations are planned to be resolved by PR #182
